indexers/dbms/PostgreSQLIndexer/__init__.py

```diff
@@ -0,0 +1,145 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import pickle
+from typing import Optional
+
+from jina.executors.indexers import BaseIndexer
+
+
+class PostgreSQLDBMSIndexer(BaseIndexer):
+    # TODO:  this class need to be a subclass from the DBMSIndexer (when it's merged into master)
+    """:class:`PostgreSQLDBMSIndexer` PostgreSQL based KV Indexer.
+        Initialize the PostgreSQLDBIndexer.
+
+        :param hostname: hostname of the machine
+        :param port: the port
+        :param username: the username to authenticate
+        :param password: the password to authenticate
+        :param database: the database name
+        :param collection: the collection name
+        :param args: other arguments
+        :param kwargs: other keyword arguments
+    """
+
+    def __init__(self,
+                 hostname: str = '127.0.0.1',
+                 port: int = 5432,
+                 username: str = 'default_name',
+                 password: str = 'default_pwd',
+                 database: str = 'default_db',
+                 table: Optional[str] = 'default_table',
+                 *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        self.hostname = hostname
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database_name = database
+        self.table = table
+
+    def __enter__(self):
```

Comment:
This is not going to work inside Runtime, inside Runtime currently Executor is not used as context manager.

See the pattern used in other Indexers (like MongoDB). U can have an internal class handling the connectivity with the actual backend. And that object can be context managed to properly handle the connections